ui/app/AppLayouts/Wallet/panels/SwapInputPanel.qml

```diff
@@ -0,0 +1,257 @@
+            PathLine {
+                x: shape.leftBottomRadius
+                y: shape.height
+            }
+            PathArc {
+                x: 0
+                y: shape.height - shape.leftBottomRadius
+                radiusX: shape.leftBottomRadius
+                radiusY: shape.leftBottomRadius
+            }
+            PathLine {
+                x: 0
+                y: shape.leftTopRadius
+            }
+            PathArc {
+                x: shape.leftTopRadius
+                y: 0
+                radiusX: shape.leftTopRadius
+                radiusY: shape.leftTopRadius
+            }
+        }
+    }
+
+    contentItem: RowLayout {
+        spacing: 20
+        ColumnLayout {
+            Layout.preferredWidth: parent.width*.6
+            Layout.fillHeight: true
+
+            AmountToSend {
+                Layout.fillWidth: true
+                id: amountToSendInput
+                objectName: "amountToSendInput"
+                caption: root.caption
+                interactive: d.isSelectedHoldingValidAsset
+                selectedHolding: d.selectedHolding
+                fiatInputInteractive: root.fiatInputInteractive
+
+                multiplierIndex: d.isSelectedHoldingValidAsset && !!holdingSelector.selectedItem && !!holdingSelector.selectedItem.decimals
+                                 ? holdingSelector.selectedItem.decimals
+                                 : 0
+
+                maxInputBalance: root.swapSide === SwapInputPanel.SwapSide.Pay ? d.maxInputBalance : Number.POSITIVE_INFINITY
+                currentCurrency: root.store.currencyStore.currentCurrency
+                formatCurrencyAmount: root.store.currencyStore.formatCurrencyAmount
+            }
+        }
+        ColumnLayout {
+            Layout.preferredWidth: parent.width*.4
+
+            Item { Layout.fillHeight: true }
+
+            HoldingSelector {
+                id: holdingSelector
+                objectName: "holdingSelector"
+                Layout.rightMargin: d.isSelectedHoldingValidAsset ? -root.padding : 0
+                Layout.alignment: Qt.AlignRight
+                Layout.preferredHeight: 38
+
+                selectedSenderAccount: root.store.selectedSenderAccount.address
+                assetsModel: root.store.processedAssetsModel
+                networksModel: root.store.flatNetworksModel
+                formatCurrentCurrencyAmount: function(balance) {
+                    return root.store.currencyStore.formatCurrencyAmount(balance, root.store.currencyStore.currentCurrency)
+                }
+                formatCurrencyAmountFromBigInt: function(balance, symbol, decimals) {
+                    return root.store.formatCurrencyAmountFromBigInt(balance, symbol, decimals)
+                }
+                onItemSelected: {
+                    d.setSelectedHoldingId(holdingId, holdingType)
+                }
+                onSearchTextChanged: root.store.assetSearchString = searchText
```

Comment:
I wonder why this is needed? cause the filteration happens in the store? we need to move the filteration this place in that case. Cause we wont be using TransactionStore here at all, so this dependency is not correct.

Reply:
Yeah it happens in the store; that's one of the reasons the components depend on it for now

Reply:
Fixed